app/scripts/lib/idStore-migrator.js

```diff
@@ -26,11 +28,10 @@ module.exports = class IdentityStoreMigrator {
     return new Promise((resolve, reject) => {
       this.idStore.submitPassword(password, (err) => {
         if (err) return reject(err)
-        try {
-          resolve(this.serializeVault())
-        } catch (e) {
-          reject(e)
-        }
+        const serialized = this.serializeVault()
+        this.checkForErrors(serialized)
```

Comment:
consider using [`denodeify`](https://github.com/matthew-andrews/denodeify) for clean callback-promise interface